R/checks.R

```diff
@@ -272,3 +272,65 @@ check_req_labelled_metadata_fields <- function(supertbl) {
     )
   }
 }
+
+
+#' @title
+#' Check an argument with checkmate
+#'
+#' @importFrom cli cli_abort
+#' @importFrom rlang caller_arg
+#'
+#' @param x An object to check
+#' @param arg The name of the argument to include in an error message. Captured
+#' by `rlang::caller_arg()` by default
+#' @param ... additional arguments passed on to checkmate
+#'
+#' @return
+#' `TRUE` if `x` passes the checkmate check. An error otherwise with the name of
+#' the checkmate function as a `class`
+#'
+#' @name checkmate
+#' @keywords internal
+NULL
+
+# Function factory to wrap checkmate functions
+#' @importFrom rlang caller_arg
+#' @importFrom cli cli_abort
+#' @noRd
+wrap_checkmate <- function(f, f_name = caller_arg(f)) {
+  function(x, ..., arg = caller_arg(x)) {
+    out <- f(x, ...)
+
+    if (isTRUE(out)) {
+      return(TRUE)
+    }
+
+    cli_abort(
+      message = c(
+        "x" = "{.arg {arg}} is invalid",
```

Comment:
> I'm wondering if the `"x"` and `"i"` part of the message should be optional arguments to the `wrap_checkmate()` function that can override the default. (Would have `NULL` as default for `"i"` but can supply more info if we want.)

To clarify, is the suggestion that we keep the `checkmate` message in `"!"` in all cases but add the ability to tack on additional lines of info to specific error messages?

Reply:
What I was thinking is that the function definition of `wrap_checkmate()` could be:

```r
wrap_checkmate <- function(f, 
                           "x" = "You've supplied {y} for {.arg {arg}} which is not a valid value.",
                           "!" = "{out}",
                           "i" = NULL)
```

The function definition inside the function factory could pass these defaults on, allowing for modification (note, have to change `x` to `y` below to avoid conflict with `x` for the cli message.

```
  function(y, "x" = x, "!" = `!`, "i" = i, ..., arg = caller_arg(x)) {
```

And then you'd pass these values down to `cli_abort()`:

```r
    cli_abort(
      message = c(
        "x" = x,
        "!" = `!`,
        "i" = i
      ),
```